### ISSUE TITLE
Table Styling

### ISSUE THREAD

**Issue by sfrisk** [jquery/css-chassis#53]
Base styling for tables

**Comment by samatdav** [comment]
There are some popular table styles. Is there something what can be used from them?

1. Bootstrap 
http://getbootstrap.com/css/#tables
2. Purecss
http://purecss.io/tables/

**Comment by geekman-rohit** [comment]
As far as Tables are concerned, I love what bootstrap does, its neat and clean and systematic
classes like table, table-striped, table-bordered makes sense.
We should start running tests for tables of different frameworks and compare maybe?
I have used bootstrap for quite a long time now, and it's tables ad buttons are what I like the most about bootstrap.

**Comment by prasunanand** [comment]
Tables must be supporting these features:
1. Bordered
2.Celled
3.Compact
4.Responsive 
5.Striped
6.Contextual rows, columns and cells(error , warning , active, disabled,icon support)

The first 5 can be achieved by adding the class to the tables.
Contextual rows and cells can be made by providing the class names to them . Also we need to provide the flexibilty of how the devs want to modify their rows or celumns or cells.

**Comment by prasunanand** [comment]
Adding to this ,The concept of "Sortable Tables" would be very useful in CRUD based web applications

**Comment by hirenmayani** [comment]
@Prasunanand, Don't you think we'll require js for 'sortable table'. And we're not going to use it in css-chassit.

**Comment by arschmitz** [comment]
Something i would like to see for tables is support for responsive table styles like in jquery mobile we have. http://demos.jquerymobile.com/1.4.5/table-reflow/ and https://github.com/jquery/jquery-mobile/issues/7700 a request for better styling on those

**Comment by prasunanand** [comment]
@arschmitz ,Responsive tables  would definetly be nice.
@hirenmayani We can use data attributes to sort tables. http://html5doctor.com/html5-custom-data-attributes/

**Comment by geekman-rohit** [comment]
@prasunanand , they are just attributes, we still need javascript to access them and do actual sorting, they provide a clearer mechanism to provide data! and we arent relying on html5 yet, I think.

**Comment by sfrisk** [comment]
I think the data attribute is closer aligned with the UI toolkit phase than css framework.  The point of Phase One is that Javascript isn't required to make anything work.

**Comment by arschmitz** [comment]
@geekman-rohit data attributes work in all browsers the only difference is how you access them.

**Comment by geekman-rohit** [comment]
@arschmitz , agreed, my point was they won't help in actual sorting without javascript, and we aren't considering javascript for phase one. But I stand corrected on the other point about not relying on html5 yet. Thanks

**Comment by MelodyCuida** [comment]
will you consider styles for cross-table?

**Comment by sfrisk** [comment]
Having `<td>` that can cover multiple rows/columns should definitely be considered in styling.

**Comment by hirenmayani** [comment]
@sfrisk are talking about rowspan and colspan by <td> that covers multiple rows/columns?

**Comment by sfrisk** [comment]
correct, @hirenmayani

**Comment by hirenmayani** [comment]
@sfrisk  what difference does it make than single row/column (designing point of view)?  Is it for making table responsive?

**Comment by sfrisk** [comment]
@hirenmayani Depends on the content of your table, but it would be important to test those cases to make sure the table is still visually pleasing.

**Comment by MichaelArestad** [comment]
Let's start simple. Let's get styles for the most basic table squared away first keeping the bordered/other variations in mind. I lean toward having the leanest styles possible. Some basic font styling and padding. Potentially a key border or two.

I absolutely do not want to go to the extent bootstrap does with all of the modifiers (though, of the two, it has the better styling).

I just put together a PR for this. https://github.com/jquery/css-chassis/pull/69

**Comment by sfrisk** [comment]
Agree on the Bootstrap point - while they do have excellent styling, I think that sometimes they offer a little too much to use.  I think the goal should be some basic styling, that would be easy for someone to build off of.
